frost-core/src/frost/round1.rs

```diff
@@ -16,21 +18,33 @@ impl<C> Nonce<C>
 where
     C: Ciphersuite,
 {
-    /// Generates a new uniformly random signing nonce.
+    /// Generates a new uniformly random signing nonce by sourcing fresh
+    /// randomness and combining with the secret key, to hedge against a bad
+    /// RNG.
     ///
     /// Each participant generates signing nonces before performing a signing
     /// operation.
     ///
-    /// An implementation of `RandomNonzeroScalar()` from the [spec].
+    /// An implementation of `nonce_generate(secret)` from the [spec].
     ///
-    /// [spec]: https://www.ietf.org/archive/id/draft-irtf-cfrg-frost-04.html#section-3.1-3.4
-    pub fn random<R>(rng: &mut R) -> Self
+    /// [spec]: https://www.ietf.org/archive/id/draft-irtf-cfrg-frost-05.html#name-nonce-generation
+    pub fn new<R>(key_package: &KeyPackage<C>, rng: &mut R) -> Self
```

Comment:
Should we just restrict this type to `Secret<C>`? This is closer to `nonce_generate()` in the spec and a little more flexible in the API.

Reply:
So I tried to just make this (or others) `Secret<C>` but of course up the callstack we need to sort the `SigningCommitments` by their identifier (index). But this one and a few others can be `Secret<C>` at least.